src/StreetNameRegistry/Municipality/ValueObjects/StreetNameHomonymAddition.cs

```diff
@@ -28,7 +28,7 @@ public StreetNameHomonymAddition(string homonymAddition, Language language)
 
         protected override IEnumerable<object> Reflect()
         {
-            yield return HomonymAddition;
+            yield return HomonymAddition.ToLowerInvariant();
```

Comment:
I've also added case insensitive logic here because I think when we are comparing two value object of this type, then we consider them the same regardless of casing.